### PR TITLE
fix Windows paths

### PR DESCRIPTION
**Pull request by cacrespo**
not elegant but solves #7 😄

### CODE REVIEW

**Comment by ultrabug** [comment]
Hi @cacrespo thanks for taking on your time to improve the project.

I manipulate `dest_path` in multiple other places like in line 101 but it does not seem to matter for your testing so I'm a bit puzzled about why this `joinpath()` alone is doing some magic :confused: 

I have no Windows to check this on, would you mind showing a real dest_path before / after example?

**Comment by cacrespo** [comment]
Hi @ultrabug, happy to be of help! 😄 

I think the problem is when the path string finish with "/" (for example in line 95). The line 101 isn't the case.
In facts, I see that there is no need  .joinpath(). Only Path() should work.

Look how works in Windows:
```
>>> from pathlib import Path
>>> a = "C:\Windows/"
>>> b = str(Path(a))

>>> print (a.replace("Windows", "") )
C:\/

>>> print (b.replace("Windows", "") )
C:\
```

This are the folders without the adjust
C:.
├───css
├───en
│   ├───topic1
│   │   └───index.en
│   └───topic2
│       └───index.en
├───fonts
├───fr
│   ├───index.fr
│   ├───topic1
│   │   └───index.fr
│   └───topic2
│       └───index.en
├───img
├───js
├───search
├───topic1
│   └───index.en
└───topic2

And now with de fix:
C:.
├───css
├───en
│   ├───topic1
│   └───topic2
├───fonts
├───fr
│   ├───topic1
│   └───topic2
├───img
├───js
├───search
├───topic1
└───topic2

**Comment by cacrespo** [comment]
> Dunno if I'm clear enough?

Yes! I will try to do it 👍

File: mkdocs_static_i18n/plugin.py [hunk 1]
```diff
@@ -118,9 +112,7 @@ def _get_non_translated_page(self, page, page_lang, config):
             )
             i18n_page.url = page.url.replace(page.name, i18n_page.name) or "."
         else:
-            i18n_page.dest_path = page.dest_path.replace(str(Path(f"{page.name}/")), "")
-            i18n_page.abs_dest_path = page.abs_dest_path.replace(str(Path(f"{page.name}/")), "")
-            i18n_page.url = page.url.replace(str(Path(f"{page.name}/")), "") or "."
+            i18n_page = self._fix_dest_paths(page)
 
         return i18n_page
```
Comment (ultrabug) [inline mkdocs_static_i18n/plugin.py:117]:
```suggestion
        return self._fix_dest_paths(i18n_page)
```

I guess it's better and safer to fix the object on return please so we're not stuck in a if / else context.

File: mkdocs_static_i18n/plugin.py [hunk 2]
```diff
@@ -140,6 +132,17 @@ def _get_page_from_paths(self, expected_paths, files):
                 f"mkdocs-static-i18n is expecting one of the following files: {expected_paths}"
             )
 
+    def _fix_dest_paths(self, page):
```
Comment (ultrabug) [inline mkdocs_static_i18n/plugin.py:135]: Just take the original `i18n_page` as input

File: mkdocs_static_i18n/plugin.py [hunk 3]
```diff
@@ -140,6 +132,17 @@ def _get_page_from_paths(self, expected_paths, files):
...
+        """
+        Corrects paths to work fine on Windows and Unix system
+        """
+        i18n_page = deepcopy(page)
```
Comment (ultrabug) [inline mkdocs_static_i18n/plugin.py:139]: Do not make a copy, we just want to modify the referenced object

File: mkdocs_static_i18n/plugin.py [hunk 4]
```diff
@@ -140,6 +132,17 @@ def _get_page_from_paths(self, expected_paths, files):
...
+        i18n_page.dest_path = page.dest_path.replace(str(Path(f"{page.name}/")), "")
```
Comment (ultrabug) [inline mkdocs_static_i18n/plugin.py:140]: please leave the `replace()` stuff where they are in the code, in this function we only want to Path() them

```suggestion
        i18n_page.dest_path = str(Path(i18n_page.dest_path))
```

Comment (ultrabug) [reply mkdocs_static_i18n/plugin.py:140]: (same for abs_dest_path)

**Review by ultrabug — CHANGES_REQUESTED** [review]
Thanks a lot, we're almost there!

I think the function should take the original i18n_page and just str(Path()) its dest_path and abs_dest_path properties before returning it

That way it will be a very simple and clean change

**Comment by cacrespo** [comment]
Hi @ultrabug! Thanks for the feedback and sorry if I can't follow you properly. My English is not very good and "my Python" is worse 😝 

I think the `replace()` is still necesary. Look the output of this test I make: 

```
    def _fix_dest_paths(self, page):
        """
        Corrects paths to work fine on Windows and Unix system
        """
        i18n_page = deepcopy(page)

        print(str(Path(i18n_page.dest_path)))
        print(i18n_page.dest_path)

        i18n_page.dest_path = page.dest_path.replace(str(Path(f"{page.name}/")), "")

        print("fix:", i18n_page.dest_path)
        print("fix:", str(Path(i18n_page.dest_path)))
```
This is the output:
```
index.fr\index.html
index.fr\index.html
fix: \index.html
fix: \index.html
topic1\index.en\index.html
topic1\index.en\index.html
fix: topic1\\index.html
fix: topic1\index.html
...
```
Only if I maintain the `replace()` the tree folder structure is built correctly. What do you think?

**Comment by ultrabug** [comment]
> Thanks for the feedback and sorry if I can't follow you properly

On the contrary, it's me feeling bad to get you changing code

> I think the replace() is still necesary

Yes, that's why I'm saying to leave the replace() as it is today in the current code

I'm only proposing to use the `_fix_dest_paths` to str(Path()) them after

**Comment by ultrabug** [comment]
If I'm not clear enough, I can put up a PR to show you what I mean

My interest here is to preserve your contribution

**Comment by cacrespo** [comment]
Hi @ultrabug! I revised all the changes and I think we can't just take the original i18n_page and returning it with the properties adjusted. Let me explain you:
In all `replace()` we search for path with forward slashes and it is the problem in Windows. For example, when in the test case  run:

`i18n_page.dest_path = page.dest_path.replace(f"{page.name}/", "")` 
on Windows the value of `i18n_page.dest_path` is `topic1\index.en\index.html` and we need replace `index.en/` (please look at "/").

This is all in if/else context before return i18n_page so if we want to apply the function fix_path on return line is necessary move all the replace hack to the function.

Another solution I tried is convert all the paths with [.as_posix()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) but didn't work (for me). Even if I try only in if/else block. 
So, I think the more simplest and cleanest solution is reconsiderate the commit b374bb0

> My interest here is to preserve your contribution

Thank you very much. You are very considerate.
I think my contribution is already made even if we finally don't merge the pull request. It is OK for me if you create and merge other PR (really I want to know how fix the problem). I can test it.

**Comment by cacrespo** [comment]
Hi @ultrabug! I revised all the changes and I think we can't just take the original i18n_page and returning it with the properties adjusted. Let me explain you:
In all `replace()` we search for path with forward slashes and it is the problem in Windows. For example, when in the test case  run:

`i18n_page.dest_path = page.dest_path.replace(f"{page.name}/", "")` 
on Windows the value of `i18n_page.dest_path` is `topic1\index.en\index.html` and we need replace `index.en/` (please look at "/").

This is all in if/else context before return i18n_page so if we want to apply the function fix_path on return line is necessary move all the replace hack to the function.

Another solution I tried is convert all the paths with [.as_posix()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) but didn't work (for me). Even if I try only in if/else block. 
So, I think the more simplest and cleanest solution is reconsiderate the commit b374bb0

> My interest here is to preserve your contribution

Thank you very much. You are very considerate.
I think my contribution is already made even if we finally don't merge the pull request. It is OK for me if you create and merge other PR (really I want to know how fix the problem). I can test it.

**Comment by cacrespo** [comment]
Hi @ultrabug! I revised all the changes and I think we can't just take the original i18n_page and returning it with the properties adjusted. Let me explain you:
In all `replace()` we search for path with forward slashes and it is the problem in Windows. For example, when in the test case  run:

`i18n_page.dest_path = page.dest_path.replace(f"{page.name}/", "")` 
on Windows the value of `i18n_page.dest_path` is `topic1\index.en\index.html` and we need replace `index.en/` (please look at "/").

This is all in if/else context before return i18n_page so if we want to apply the function fix_path on return line is necessary move all the replace hack to the function.

Another solution I tried is convert all the paths with [.as_posix()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) but didn't work (for me). Even if I try only in if/else block. 
So, I think the more simplest and cleanest solution is reconsiderate the commit b374bb0

> My interest here is to preserve your contribution

Thank you very much. You are very considerate.
I think my contribution is already made even if we finally don't merge the pull request. It is OK for me if you create and merge other PR (really I want to know how fix the problem). I can test it.

**Comment by ultrabug** [comment]
Hi mate, please see #13

**Comment by cacrespo** [comment]
Hello @ultrabug!
In a few days I will see this. 😄

**Comment by ultrabug** [comment]
fixed in #13

**Pull request closed without merging**